File: new/src/MyClass.ts
class MyClass {
    private readonly instance: any;

    constructor(module: any) {
        // @ts-ignore
        this.instance = new module.MyClass();
    }

    multiply(a: number, b: number): number {
        return this.instance.multiply(a, b);
    }

    sayHello(name: string): string {
        return this.instance.sayHello(name);
    }

    getDefaultRubberbandEngineVersion(): number {
        return this.instance.getDefaultRubberbandEngineVersion();
    }
}
export {MyClass}

File: wasm/src/MyClass.cpp
//
// Created by Tobias Hegemann on 26.09.22.
//

#include "MyClass.h"
MyClass::MyClass() {

}
float MyClass::multiply(float a, float b) {
  return a * b;
}
std::string MyClass::sayHello(std::string name) {
  return "Hello " + name;
}
int MyClass::getDefaultRubberbandEngineVersion() {
  auto *rb = new RubberBand::RubberBandStretcher(44100, 1);
  auto version = rb->getEngineVersion();
  delete rb;
  return version;
}
